htmd/parameterization/esp.py

```diff
@@ -103,6 +190,169 @@ def _generate_points(molecule, vdw_radii=[1.4, 1.6, 1.8, 2.0, 2.2], density=10):
 
         return points
 
+    def __init__(self):
+
+        self.molecule = None
+        self.qm_results = None
+        self.fixed = []
+
+        self._reciprocal_distances = None
+
+    @property
+    def ngroups(self):
+        """Number of charge groups"""
+
+        return len(self.molecule._equivalent_atom_groups)
+
+    def _map_groups_to_atoms(self, group_charges):
+
+        charges = np.zeros(self.molecule.natoms)
+        for atom_group, group_charge in zip(self.molecule._equivalent_atom_groups, group_charges):
+            charges[atom_group] = group_charge
+
+        return charges
+
+    def _compute_constraint(self, group_charges, _):
```

Comment:
What's the `_` for as last argument?

Reply:
For gradients, which aren't used in this case.

Reply:
then I don't understand what it is doing there